qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -186,6 +190,18 @@ def _validate(self, qobj, backend_options, noise_model):
         """Validate the qobj, backend_options, noise_model for the backend"""
         pass
 
+    def _save_job(self, result, backend_options):
+        if (backend_options is not None) and ("path_to_save" in backend_options):
+            path_to_save = backend_options["path_to_save"]
+            if not path_to_save.endswith("/"):
+                path_to_save += "/"
+            filename = path_to_save + "_".join(
+                [self.name(), str(result.job_id), str(int(time.time()))]
+            )
+            os.makedirs(os.path.dirname(filename), exist_ok=True)
+            with open(filename, "wb") as file:
+                pickle.dump(result, file, 0)
```

Comment:
Do not use pickle as a general purpose serialization format. As you had already mentioned in the issue this is a big security issue. This kind of use case is the exact thing you should **not** be using pickle for.

It is not hard to serialize a results object there is already a `to_dict()` method on it which gets you pretty close, you can use json, yaml, or any other number of other off the shelf serialization formats that work with python dicts. There are just a couple of types (like numpy arrays) in the dict that you'll have to handle in whatever you pick. If you go json for example it's easy to write a small encoder class to handle that.

Reply:
Got it. I also just noticed the `AerJSONEncoder` class that's already been written, so I'll try and leverage that.